clarifying description and code for static block 'var' hoisting (#35793)
mdn/content: files/en-us/web/javascript/reference/classes/static_initialization_blocks/index.md

These are [evaluated](/en-US/docs/Web/JavaScript/Reference/Classes#evaluation_order), along with any interleaved static field initializers, in the order they are declared.
Any static initialization of a super class is performed first, before that of its sub classes.

> The scope of the variables declared inside the static block is local to the block. This includes `var`, `function`, `const`, and `let` declarations. `var` declarations _do_ hoist, but only within the block.

Reviewer: ```suggestion
The scope of the variables declared inside the static block is local to the block. This includes `var`, `function`, `const`, and `let` declarations. `var` declarations will not be hoisted out of the static block.
```

What about this? This avoids talking about things like function declarations nested within static blocks.
